src/EditorFeatures/Core/Implementation/IntelliSense/Helpers.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+
+using Microsoft.VisualStudio.Text.Adornments;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense
+{
+    internal static class Helpers
+    {
+        internal static ClassifiedTextElement BuildClassifiedTextElement(ImmutableArray<TaggedText>? taggedTexts)
```

Comment:
taking a nullable immutable array is super weird since ImmutableArray already supports the concept of 'null' to begin with.  Can you just take a regular ImmutableArray, then in the code call .NullToEmpty on it if you want to deal with null arrays in a consistent fashion.

Reply:
this feels like the wrong namespace for this code.  But i don't have anything much better.

Reply:
I also could not find a better one. So, let us keep it here until we will find a better one.

---
In reply to: [232509197](https://github.com/dotnet/roslyn/pull/29016#discussion_r232509197) [](ancestors = 232509197)